core/src/main/java/org/elasticsearch/search/fetch/subphase/InnerHitsContext.java

```diff
@@ -127,14 +126,14 @@ public static void intersect(Weight weight, Weight innerHitQueryWeight, Collecto
             return;
         }
         // use random access since this scorer will be consumed on a minority of documents
-        Scorer scorer = scorerSupplier.get(true);
+        Scorer scorer = scorerSupplier.get(Long.MAX_VALUE);
```

Comment:
You could use the real cost of the `innerHitQueryWeight` or 0 ? With this modification the scorer supplier would prefer queries with advance capabilities which is the opposite of the comment above ?

Reply:
Thanks for noticing this! I'll use a leadCost of `0`.